plugins/wrappers/engine/enginewrapper.go

```diff
@@ -0,0 +1,468 @@
+	}
+}
+func utilsToGethHeader(header *ptypes.Header) *types.Header {
+	if header == nil { return nil }
+	return &types.Header{
+		ParentHash: common.Hash(header.ParentHash),
+		UncleHash: common.Hash(header.UncleHash),
+		Coinbase: common.Address(header.Coinbase),
+		Root: common.Hash(header.Root),
+		TxHash: common.Hash(header.TxHash),
+		ReceiptHash: common.Hash(header.ReceiptHash),
+		Bloom: types.Bloom(header.Bloom),
+		Difficulty: header.Difficulty,
+		Number: header.Number,
+		GasLimit: header.GasLimit,
+		GasUsed: header.GasUsed,
+		Time: header.Time,
+		Extra: header.Extra,
+		MixDigest: common.Hash(header.MixDigest),
+		Nonce: types.BlockNonce(header.Nonce),
+		BaseFee: header.BaseFee,
+		WithdrawalsHash: (*common.Hash)(header.WithdrawalsHash),
+	}
+}
+
+func gethToUtilsTransactions(transactions []*types.Transaction) []*ptypes.Transaction {
+	if transactions == nil { return nil }
+	txs := make([]*ptypes.Transaction, len(transactions))
+	for i, tx := range transactions {
+		bin, err := tx.MarshalBinary()
+		if err != nil { panic (err) }
+		txs[i] = &ptypes.Transaction{}
+		txs[i].UnmarshalBinary(bin)
+	}
+	return txs
+}
+
+func gethToUtilsHeaders(headers []*types.Header) []*ptypes.Header {
+	if headers == nil { return nil }
+	pheaders := make([]*ptypes.Header, len(headers))
+	for i, header := range headers {
+		pheaders[i] = gethToUtilsHeader(header)
+	}
+	return pheaders
+}
+
+func gethToUtilsReceipts(receipts []*types.Receipt) []*ptypes.Receipt {
+	if receipts == nil { return nil }
+	preceipts := make([]*ptypes.Receipt, len(receipts))
+	for i, receipt := range receipts {
+		preceipts[i] = &ptypes.Receipt{
+			Type: receipt.Type,
+			PostState: receipt.PostState,
+			Status: receipt.Status,
+			CumulativeGasUsed: receipt.CumulativeGasUsed,
+			Bloom: ptypes.Bloom(receipt.Bloom),
+			Logs: gethToUtilsLogs(receipt.Logs),
+			TxHash: core.Hash(receipt.TxHash),
+			ContractAddress: core.Address(receipt.ContractAddress),
+			GasUsed: receipt.GasUsed,
+			BlockHash: core.Hash(receipt.BlockHash),
+			BlockNumber: receipt.BlockNumber,
+			TransactionIndex: receipt.TransactionIndex,
+		}
+	}
+	return preceipts
+}
+
+func gethToUtilsWithdrawals(withdrawals []*types.Withdrawal) []*ptypes.Withdrawal {
+	if withdrawals == nil { return nil }
+	pwithdrawals := make([]*ptypes.Withdrawal, len(withdrawals))
+	for i, withdrawal := range withdrawals {
+		pwithdrawals[i] = &ptypes.Withdrawal{
+			Index: withdrawal.Index,
+			Validator: withdrawal.Validator,
+			Address: core.Address(withdrawal.Address),
+			Amount: withdrawal.Amount,
+		}
+	}
+	return pwithdrawals
+}
+
+func gethToUtilsBlock(block *types.Block) *ptypes.Block {
+	if block == nil { return nil }
+	return ptypes.NewBlockWithHeader(gethToUtilsHeader(block.Header())).WithBody(gethToUtilsTransactions(block.Transactions()), gethToUtilsHeaders(block.Uncles())).WithWithdrawals(gethToUtilsWithdrawals(block.Withdrawals()))
+	
+	// NewBlockWithWithdrawals(
```

Comment:
Let's get rid of this comment block and the one below

Reply:
Removed all unnecessary comments.